presto-main/src/main/java/io/prestosql/operator/WindowOperator.java

```diff
@@ -607,6 +623,70 @@ public TransformationState<Page> process(WindowPartition partition)
         }
     }
 
+    private class FixedSizeWindowPartitionsToOutputPages
+            implements Transformation<WindowPartition, Page>
+    {
+        BlockBuilder windowBlockBuilder;
+        final PageBuilder pageBuilder;
+        final FramedWindowFunction framedFunction;
+        final FixedSizeWindowFunction function;
+
+        FixedSizeWindowPartitionsToOutputPages(FramedWindowFunction framedFunction)
+        {
+            windowBlockBuilder = outputTypes.get(outputTypes.size() - 1).createBlockBuilder(null, 1024);
+            pageBuilder = new PageBuilder(outputTypes);
+            this.framedFunction = framedFunction;
+            this.function = ((FixedSizeWindowFunction) framedFunction.getFunction());
+        }
+
+        @Override
+        public TransformationState<Page> process(WindowPartition partition)
+        {
+            boolean finishing = partition == null;
+            if (finishing) {
+                if (windowBlockBuilder.getPositionCount() == 0) {
+                    if (pageBuilder.isEmpty()) {
+                        return TransformationState.finished();
+                    }
+                }
+                else {
+                    int[] positions = function.selectPositions(windowBlockBuilder.getPositionCount());
+                    copyPositions(partition, positions);
+                }
+
+                // Output the remaining page if we have anything buffered
+                Page page = pageBuilder.build();
+                pageBuilder.reset();
+                return TransformationState.ofResult(page, false);
+            }
+
+            // process whole partition without output channels for unwanted copy and memory usage
+            while (partition.hasNext()) {
+                partition.processFixedSizeWindowNextRow(framedFunction, windowBlockBuilder);
+            }
+            int[] positions = function.selectPositions(windowBlockBuilder.getPositionCount());
+            copyPositions(partition, positions);
```

Comment:
ok. It looks good. So this part is 
- Update the reservoir buffer with processFixedSizeWindowNextRow
- Get the selected positions
- Copy the rows at the selected positions